iothub/device/src/Authentication/IotHubConnectionStringBuilder.cs

```diff
@@ -109,28 +100,22 @@ public IotHubConnectionStringBuilder(string iotHubConnectionString)
         /// SAS token, when that token expires, the client must be disposed, and if desired, recreated
         /// with a newly derived SAS token.
         /// </remarks>
-        public string SharedAccessSignature { get; internal set; }
+        public string SharedAccessSignature { get; set; }
 
         /// <summary>
-        /// Gets or sets the authentication method to be used with the IoT hub service.
-        /// </summary>
-        public IAuthenticationMethod AuthenticationMethod { get; }
-
-        /// <summary>
-        /// Indicates if the connection string indicates if an x509 certificate is specified for authentication.
+        /// The client X509 certificates used for authenticating with IoT hub.
         /// </summary>
-        public bool UsingX509Cert { get; internal set; }
+        public X509Certificate2 Certificate { get; set; }
```

Comment:
Why does a "connection string builder" need cert info?

Reply:
This class holds the authentication information for any client to connect to IoT hub. It had been poorly named in the past. The next PR addresses the renaming: https://github.com/Azure/azure-iot-sdk-csharp/pull/2609